cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -50,28 +102,409 @@ CDAP supports logging through standard `SLF4J (Simple Logging Facade for Java)
+programs of an application and system services with the logging framework:
+
+  - Logs are collected from individual programs running in YARN containers.
+
+  - YARN containers have their own local log files.
+
+  - CDAP System Services run (depending on the service) either on cluster edge nodes or in
+    YARN containers. Where they run determines the file that configures that service's
+    logging.
+
+  - CDAP logs are published to Kafka topics.
+
+  - The Log Saver Service (*Log.saver*) is configured to read log messages for the
+    ``logs.user-v2`` Kafka topic. The number of log saver instances can be scaled to process
+    the Kafka partitions in parallel, if needed. Log.saver reads the messages from Kafka,
+    buffers and sorts them in-memory, before persisting them to HDFS. These messages are
+    written to files corresponding to the programs the log messages belong to.
+
+    *Note:* These files are configured to rotate based on time and size; they can be
+    changed using the properties ``log.pipeline.cdap.file.max.size.bytes`` and
+    ``log.pipeline.cdap.file.max.lifetime.ms`` in the :ref:`cdap-site.xml
+    <appendix-cdap-default-logging>` file as described in 
+    :ref:`logging-monitoring-logging-pipeline-configuration`.
+
+  - Custom pipelines can be configured by adding an XML file in a prescribed location.
+    Each pipeline buffers log messages in-memory and sorts them based on their timestamp. 
+   
+  - Metrics are created based on the results of the pipelines, and passed to the Metrics
+    Service.
+   
+  - For security, the files written out to persistent storage in HDFS have permissions set
+    such that they are accessible only by the ``cdap`` user.
+
+  - Log.saver also emits metrics for each *info, debug,* or *error* message received. This
+    helps in monitoring the amount of logs emitted by applications, application's programs,
+    system services, etc. and can be retrieved by querying the :ref:`CDAP metrics system
+    <http-restful-api-metrics>`.
+
+.. figure:: /_images/logging/logging-framework.png
+    :figwidth: 100%
+    :width: 800px
+    :align: center
+
+    **CDAP Logging Framework:** From YARN containers, through Kafka and the Log Saver
+    Service, to HDFS
+
+Logging Standards
+-----------------
+Logging uses the standard `SLF4J (Simple Logging Facade for Java)
+<http://www.slf4j.org/manual.html>`__ APIs and `Logback
+<https://logback.qos.ch/manual>`__. Logging is configured using instances of Logback's
+"logback" file, consisting of *logging pipelines* with *logging appenders*:
+
+  - A **logging pipeline** is a process that consumes log events from Kafka, invokes
+    *logback appenders* defined in its configuration, buffering, sorting, and then
+    persisting the log messages.
+
+  - A **logback appender** (or *appender*) is a Java class, responsible for writing the
+    log events to persistent storage and maintaining metadata about that storage.
```

Comment:
not necessarily persisted. What it really does is consume and process messages, which typically includes/implies persisting them; however it could also collect metrics, or emit alerts when it finds certain messages.